packages/app/context/TokenListContext.tsx

```diff
@@ -0,0 +1,111 @@
+"use client";
+
+import {
+  Dispatch,
+  PropsWithChildren,
+  SetStateAction,
+  createContext,
+  useCallback,
+  useContext,
+  useEffect,
+  useState,
+} from "react";
+import { useNetwork } from "wagmi";
+import defaultGnosisTokenlist from "public/assets/blockchains/gnosis/tokenlist.json";
+import defaultEthereumTokenlist from "public/assets/blockchains/ethereum/tokenlist.json";
+
+import { Token, TokenFromTokenlist } from "@/models/token/types";
+
+const GNOSIS_CHAIN_ID = 100;
+
+const DEFAULT_TOKEN_LIST_BY_CHAIN: {
+  [chainId: number]: TokenFromTokenlist[];
+} = {
+  1: defaultEthereumTokenlist,
+  100: defaultGnosisTokenlist,
+};
+
+const TOKEN_LIST_BY_CHAIN_URL: { [chainId: number]: string } = {
+  1: "https://tokens.1inch.eth.link/",
+  100: "https://tokens.honeyswap.org/",
+};
+
+const TokenListContext = createContext<{
+  tokenList: TokenFromTokenlist[];
+  getTokenLogoURL?: (tokenAddress: string) => string;
+  getTokenFromList?: (tokenAddress: string) => TokenFromTokenlist | undefined;
+}>({
+  tokenList: DEFAULT_TOKEN_LIST_BY_CHAIN[GNOSIS_CHAIN_ID],
+});
+
+const mergeTokenlists = (
+  defaultTokenList: TokenFromTokenlist[],
+  tokenlist: TokenFromTokenlist[]
+) =>
+  defaultTokenList.concat(
+    tokenlist.filter(
+      (token) =>
+        !defaultTokenList.some(
+          (defaultToken) => defaultToken.address === token.address
+        )
+    )
+  );
+
+export const TokenListProvider = ({ children }: PropsWithChildren) => {
+  const { chain } = useNetwork();
+
+  const defaultTokenList = chain
+    ? DEFAULT_TOKEN_LIST_BY_CHAIN[chain.id]
+    : DEFAULT_TOKEN_LIST_BY_CHAIN[GNOSIS_CHAIN_ID];
```

Comment:
I think defaulting to Gnosis chain will bring unexpected behaviour. But we can't test it write now.

Reply:
1 solution would be to return an empty state if there is no chain.

Reply:
but we do want a default chain, no?

For eg:

I want to try out Stackly without connecting, which network should be the default for showing tokens?  (token picker), etc.

I think that network, for us, will be gnosis.